### PR TITLE
Initial version

### PR DESCRIPTION
**Pull request by snork-nbl**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,116 @@
-# AlexaSmartHomeSkillAdapater
\ No newline at end of file
+# AlexaSmartHomeSkillAdapter
+
+The library provides convenient way to implement [Alexa Smart Home Skill Adapter](https://developer.amazon.com/public/solutions/alexa/alexa-skills-kit/overviews/understanding-the-smart-home-skill-api). It takes care about JSON verification, data parsing and response formating.
```
Comment (snork-nbl) [inline README.md]: Strictly speaking the library is not for the Smart Home Skill Adapter implementation. It leverages the Adapter, which is an essential part of the architecture.

File: README.md [hunk 2]
```diff
@@ -1 +1,116 @@
...
+mySkill <- AlexaSmartHomeSkillAdapter();
+```
+
+### Instance Methods
+
+#### registerAppliance(applianceId, applianceInfo, actionCallback)
+
+The _registerAppliance()_ methods makes the devices be accessible through [Alexa Smart Home Skill API](https://developer.amazon.com/public/solutions/alexa/alexa-skills-kit/overviews/understanding-the-smart-home-skill-api) requests. _applianceId_ parameter sets a device identifier. It must be unique across all devices accessible through this adapter. _applianceInfo_ is a __table__ with a number of [mandatory properties](https://developer.amazon.com/public/solutions/alexa/alexa-skills-kit/docs/smart-home-skill-api-reference#discoverappliancesresponse), excluding _applianceId_ and _isReachable_. The library itself uses only _actions_ property to verify incoming request against list of device supported actions.
+```squirrel
+local smartToasterInfo = {
```
Comment (snork-nbl) [inline README.md]: `smartToasterInfo` has never been used in the example.

File: README.md [hunk 3]
```diff
@@ -1 +1,116 @@
...
+    "version": "1.0"
+};
+// Callback executed upon Alexa request
+local cb = function(appId, action, details) {
+	server.log("Got action " + action + " for " + appId + " with " + http.jsonencode(details));
+	device.send("Control", action);
+}
+
+local alexa = AlexaSmartHomeSkill(http);
+alexa.registerAppliance("SmartToaster", smartToasterApp, cb);
```
Comment (snork-nbl) [inline README.md]: `smartToasterApp` is used but has never been defined.

File: AlexaSmartHomeSkill.nut [hunk 4]
```diff
@@ -0,0 +1,1089 @@
...
+
+    // Debug flag
+    _debug = true;
+
+     // Constructor
+     // Parameters:
+     //         httpTransport  an instance of http based class
+     // Return:
+     //         Nothing
+    constructor(httpTransport = http) {
```
Comment (snork-nbl) [inline AlexaSmartHomeSkill.nut]: This http constructor parameter should be there just for a testing purpose. Please specify the problem you are trying to solve?

File: README.md (hunk 1 above)
Comment (sergey-nbl) [inline README.md]: Nope, Smart Home Skill (API) is part of of Alexa service. An adapter is a layer that accepts and converts Skill message into the form know by enpoint device.

File: README.md (hunk 3 above)
Comment (sergey-nbl) [inline README.md]: Thanks, fixed

File: README.md (hunk 2 above)
Comment (sergey-nbl) [inline README.md]: Thanks, fixed

File: AlexaSmartHomeSkill.nut (hunk 4 above)
Comment (sergey-nbl) [inline AlexaSmartHomeSkill.nut]: I do not solve any problem, I add just a bit more flexibility with almost zero overhead

**Comment by sergey-nbl** [comment]
Docs are not ready yet, as well as sanity tests.

**Pull request merged**